addons/mail/static/tests/systray/systray_messaging_menu_tests.js

```diff
@@ -782,5 +782,48 @@ QUnit.test('global counter with channel previews', function (assert) {
     messagingMenu.destroy();
 });
 
+QUnit.test('messaging menu widget: expand on thread preview', function (assert) {
+    assert.expect(3);
+
+    var messagingMenu = new MessagingMenu();
+    var taskMessage = [{
+        author_id: [1, "Demo"],
+        body: "<p>Office Design Project</p>",
+        res_id: 123,
+        model: 'some.res.model',
+        needaction: true,
+        needaction_partner_ids: [44],
+    }];
+    this.data['mail.message'].records =
+        this.data['mail.message'].records.concat(taskMessage);
+
+    testUtils.mock.addMockEnvironment(messagingMenu, {
+        services: this.services,
+        data: this.data,
+        session: {
+            partner_id: 44,
+        },
+        intercepts: {
+            do_action: function (ev) {
+                assert.deepEqual(ev.data.action, {
+                    res_id: 123,
+                    res_model: 'some.res.model',
+                    type: 'ir.actions.act_window',
+                    views: [[false, 'form']],
+                }, "should open the document in form view");
+            },
+        },
+    });
+    messagingMenu.appendTo($('#qunit-fixture'));
+
+    testUtils.dom.click(messagingMenu.$('.dropdown-toggle'));
+
+    var $preview = messagingMenu.$('.o_mail_preview');
+    assert.containsOnce(messagingMenu, '.o_thread_window_expand', "should display one preview");
+    assert.strictEqual($preview.data('document-model'), "some.res.model", "preview should be from crm.lead");
+    $preview.find('.o_thread_window_expand').click();
+
```

Comment:
add `assert.step` and `assert.verifySteps` after `click`.
If there is an error, it's easier to debug if the error message tells that the step `do_action` is missing than having a mismatch of assertion count.